packages/webpack/src/runtime/runtimeBuilder.js

```diff
@@ -1,11 +1,55 @@
 const { RUNTIME_KEY } = require('../ENUM.json')
 const diag = require('../buildtime/diagnostics.js')
-const { assembleRuntime } = require('./assemble.js')
+const { assembleRuntime, prepareSource } = require('./assemble.js')
 const path = require('node:path')
 
-/** @import {LavaMoatPluginOptions} from '../buildtime/types' */
+const { RuntimeModule } = require('webpack')
+
+class VirtualRuntimeModule extends RuntimeModule {
+  /**
+   * @param {Object} options - The options for the VirtualRuntimeModule.
+   * @param {string} options.name - The name of the module.
+   * @param {string} options.source - The source code of the module.
+   * @param {number} [options.stage] - The stage of runtime. One of
+   *   RuntimeModule.STAGE_*.
+   * @param {boolean} [options.withoutClosure] - Make the source code run
+   *   outside the closure for a runtime module
+   */
+  constructor({
+    name,
+    source,
+    stage = RuntimeModule.STAGE_NORMAL,
+    withoutClosure = false,
+  }) {
+    super(name, stage)
+    this.withoutClosure = withoutClosure
+    this.virtualSource = `;${source};`
+  }
+  shouldIsolate() {
+    return !this.withoutClosure
+  }
+
+  generate() {
```

Comment:
(suggestion)

tag methods with `@override` if they are overrides of the superclass